### ISSUE TITLE
Bug with checker: errors not reported

### ISSUE THREAD

**Issue by jbondc** [Microsoft/TypeScript#2053]
```javascript
interface obj {
   a : string
}

interface Foo {
	[x: string]: obj
}
var t: Foo
t['dd'] = 'hey' // no error
```
The error is 'checked' but not reported because of:
https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L7162

Not clear what's going on there

**Comment by jbondc** [comment]
Hmm, or close if can't reproduce it's working now :/

**Comment by vladima** [comment]
not repro on the latest bits.
